pennylane/transforms/decompositions/clifford_t/solovay_kitaev.py

```diff
@@ -339,9 +341,8 @@ def _solovay_kitaev(umat, n, u_n1_ids, u_n1_mat):
     [map_tape], _ = qml.map_wires(new_tape, wire_map={0: op.wires[0]}, queue=True)
 
     # Get phase information based on the decomposition effort
-    global_phase = qml.GlobalPhase(
-        approx_set_gph[index] - gate_gph if depth or qml.math.isclose(gate_gph, 0.0) else 0.0
-    )
+    phase = approx_set_gph[index] - gate_gph if depth or qml.math.allclose(gate_gph, 0.0) else 0.0
+    global_phase = qml.GlobalPhase(qml.math.array(phase, like=interface))
```

Comment:
Is the global phase only supposed to be appended if needed? I'm seeing it here for example (with the parameter equal to zero):
```python
import pennylane as qml

with qml.tape.QuantumTape() as tape:
    qml.CNOT([0, 1])

(tape,), func = qml.clifford_t_decomposition(tape)

print(tape.draw())
```

Reply:
It's being done for consistency so that the final operation from the transformed circuit will always be a `qml.GlobalPhase`.

Reply:
I think it's better to leave the global phase out, if zero, since it's easier to understand the output circuit and not have to carry around an unnecessary op.

Reply:
I have added a [check](https://github.com/PennyLaneAI/pennylane/blob/b89838f93fc00a1d6116a5062ea86a7d3e4a3e4f/pennylane/transforms/decompositions/clifford_t/clifford_t_transform.py#L485) in the transform that should take care of it.